src/main/java/picard/analysis/CompareMetrics.java

```diff
@@ -50,35 +70,244 @@
+                        differences.add(diffString);
+                        log.error(diffString);
+                    }
+                }
+            }
+        }
+
+        if (!IGNORE_HISTOGRAM_DIFFERENCES) {
+            final boolean histogramsEqual = mf1.areHistogramsEqual(mf2);
+            if (!histogramsEqual) {
+                final String diffString = "Metric Histograms differ";
+                differences.add(diffString);
+                log.error(diffString);
+            }
+            if (retVal == 0 && !histogramsEqual) {
+                retVal = 1;
+            }
+        }
+        if (OUTPUT != null) {
+            final String header = "Comparison of " + mf1.getMetrics().get(0).getClass().getName() + " metrics between "
+                    + metricFile1.getAbsolutePath() + " and " + metricFile2.getAbsolutePath() + "\nResult = " +
+                    ((retVal == 0) ? "Pass" : "Fail");
+            writeTextToFile(OUTPUT, header, differences);
+        }
+
+        return retVal;
+    }
+
+    protected SimpleResult compareMetricValues(final Object value1, final Object value2, final String metricName) {
+        boolean equal = true;
+        String description = "";
+        if (value1 == null || value2 == null) {
+            if (value1 != null || value2 != null) {
+                equal = false;
+                description = "One of the values is null";
+            }
+        } else {
+            if (value1 instanceof Number) {
+                double numValue1 = ((Number) value1).doubleValue();
+                double numValue2 = ((Number) value2).doubleValue();
+                double relativeChange = (numValue1 == 0 && numValue2 == 0) ? 0 : (numValue1 == 0) ? Double.MAX_VALUE : (numValue1 - numValue2) / numValue2;
+                if (relativeChange != 0) {
+                    if (MetricToAllowableRelativeChange.containsKey(metricName)) {
+                        double allowableRelativeChange = MetricToAllowableRelativeChange.get(metricName);
+                        if (abs(relativeChange) >= allowableRelativeChange) {
+                            equal = false;
+                            description = "Changed by " + RELATIVE_CHANGE_DECIMAL_FORMAT.format(relativeChange) +
+                                    " which is outside of the allowable tolerance of " + RELATIVE_CHANGE_DECIMAL_FORMAT.format(allowableRelativeChange);
+                        } else {
+                            equal = true;
+                            description = "Changed by " + RELATIVE_CHANGE_DECIMAL_FORMAT.format(relativeChange) +
+                                    " which is within the allowable tolerance of " + RELATIVE_CHANGE_DECIMAL_FORMAT.format(allowableRelativeChange);
+                        }
+                    } else {
+                        equal = false;
+                        description = "Changed by " + RELATIVE_CHANGE_DECIMAL_FORMAT.format(relativeChange);
```

Comment:
It's probably fine, but do you think it's OK that using a fixed format in this way will prevent some differences from showing up? E.g. from the test run output I see:
```
ERROR	2020-03-30 14:51:21	CompareMetrics	Row: 0 Metric: GENOME_TERRITORY values differ. Value1: 2745186694 Value2: 2745186691 Changed by 0.0000000
ERROR	2020-03-30 14:51:21	CompareMetrics	Row: 0 Metric: PCT_EXC_BASEQ values differ. Value1: 0.122134 Value2: 0.122133 Changed by 0.0000082
ERROR	2020-03-30 14:51:21	CompareMetrics	Row: 1 Metric: GENOME_TERRITORY values differ. Value1: 2745186694 Value2: 2745186691 Changed by 0.0000000
ERROR	2020-03-30 14:51:21	CompareMetrics	Row: 1 Metric: PCT_EXC_BASEQ values differ. Value1: 0.122134 Value2: 0.122133 Changed by 0.0000082
```

Reply:
I went back and forth on that.  I agree that it's losing information, so I'll put it back.